api/provider/aws/test_aws/environment_delete_test.go

```diff
@@ -117,3 +118,48 @@ func TestDeleteEnvironmentIdempotence(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestCheckEnvironmentDependencies(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	cases := []string{"load_balancer", "service", "task"}
+	for _, c := range cases {
+		tagStore := tag.NewMemoryStore()
+		tags := []models.Tag{
+			models.Tag{
+				EntityID:   "env_id",
+				EntityType: "environment",
+				Key:        "name",
+				Value:      "env_name",
+			},
+			models.Tag{
+				EntityID:   "id",
+				EntityType: c,
+				Key:        "environment_id",
+				Value:      "env_id",
+			},
+		}
+
+		for _, tag := range tags {
+			if err := tagStore.Insert(tag); err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		target := provider.NewEnvironmentProvider(mockAWS.Client(), tagStore, mockConfig)
+		err := target.Delete("env_id")
+		if serverError, ok := err.(*errors.ServerError); ok {
+			if serverError.Code != errors.DependencyError {
+				t.Errorf("Expected DependencyError for entity type %s\nActual Error: %s", c, err)
```

Comment:
it would probably be easier to use `assert` pkg here:
```go
err := target.Delete("env_id")
if err, ok := err.(*errors.ServerError); ok {
    assert.Equal(t, errors.DependencyError, err.Code)
} else {
    t.Fatalf("Error was not of type *errors.ServerError: %#v", err)
}
```

note that using `%#v` will show the object and its type

Reply:
Yeah that reads better. Any obvious advantage for choosing one over the other?

Reply:
If you mean the `assert` package, it's less lines of code which tends to make things more readable. 

If you mean `%s` vs `%#v`, the latter gives more detailed output when the test fails, something like:
```
--- FAIL: TestThing (0.00s)
        some_test.go:6: Error was not of type *errors.ServerError: &errors.errorString{s:"foo"}
```
vs
```
--- FAIL: TestThing (0.00s)
        some_test.go:6 Assertion to ServerError failed
            Error: foo
```